src/app/application.cpp

```diff
@@ -771,6 +795,64 @@ void Application::shutdownCleanup(QSessionManager &manager)
 }
 #endif
 
+#ifdef Q_OS_WIN
+namespace
+{
+    std::uint64_t autoMemoryWorkingSetSize(const std::uint64_t memorySize)
+    {
+        const std::uint64_t GB = 1024 * 1024 * 1024;
+
+        return (memorySize <= (4 * GB))
+                ? memorySize / 2
+                : (2 * GB) + (memorySize / 10);
+    }
+}
+
+void Application::applyMemoryWorkingSetLimit()
+{
+    MEMORYSTATUSEX statex;
+    statex.dwLength = sizeof(statex);
+    if (!::GlobalMemoryStatusEx(&statex))
+    {
+        const DWORD errorCode = ::GetLastError();
+        QString reason = tr("Error %1.").arg(errorCode);
```

Comment:
`Error code: %1`

Reply:
> `Error code: %1`

`Reason: Error code: 1234.` are you sure?

Reply:
No, I mean: `Failed to set Memory Working Set limit. Error code: 1234. Error message: "function failure blah blah"`

Reply:
>If you insist on showing the error code anyway

Sometimes I think this will help the developers when reading logs sent from users. We can look up the error code on msdn and not rely on localized error messages.